contracts/mixins/ProtocolTokenHandler.sol

```diff
@@ -0,0 +1,138 @@
+pragma solidity >=0.5.0 <0.6.0;
+pragma experimental ABIEncoderV2;
+
+import "../interfaces/IERC20.sol";
+import "../openzeppelin/Address.sol";
+import "../openzeppelin/SafeERC20.sol";
+import "../openzeppelin/Ownable.sol";
+
+contract ProtocolTokenHandler is Ownable {
+    using Address for address;
+    using SafeERC20 for IERC20;
+
+    address public protocolTokenAddress;
+    uint256 public requiredCount = 2;
+    mapping (address => bool) public isSigner;
+    mapping (address => uint256) public userNonce;
+
+    event AddSigner(address indexed signer);
+    event RemoveSigner(address indexed signer);
+    event SetRequiredCount(uint256 indexed requiredCount);
+    event Deposit(address indexed sender, uint256 indexed amount);
+    event Withdraw(address indexed caller, address indexed recipient, uint256 indexed amount);
```

Comment:
`uint` parameters of events should rarely be `indexed`. `address` parameters are mostly indexed.
think of it as if "do you want to find all events with this particular argument value?".
every indexed parameter costs you a bit of extra gas